src/Admin/Filter/CategoryFilter.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Admin\Filter;
+
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\Type\Filter\DefaultType;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
+use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\DoctrineORMAdminBundle\Filter\Filter;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+final class CategoryFilter extends Filter
+{
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    public function __construct(CategoryManagerInterface $categoryManager)
+    {
+        $this->categoryManager = $categoryManager;
+    }
+
+    /**
+     * @param string            $alias
+     * @param string            $field
+     * @param array|string|null $data
+     */
+    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data): void
+    {
+        if (null === $data || !\is_array($data) || !\array_key_exists('value', $data)) {
+            return;
+        }
+
+        if ($data['value']) {
+            $queryBuilder
+                ->andWhere(sprintf('%s.%s = :category', $alias, $field))
+                ->setParameter('category', $data['value'])
+            ;
+        }
+
+        $this->active = null !== $data['value'];
+    }
+
+    public function getDefaultOptions(): array
```

Comment:
```suggestion
    /**
    * @return array<string, null>
    */
    public function getDefaultOptions(): array
```

Shall we add those for further static code analysis support?